runtime/src/iree/modules/vmvx/module.c

```diff
@@ -639,47 +639,42 @@ IREE_VMVX_ABI_DEFINE_SHIM(mmt4d, v);
 
 IREE_VMVX_ABI_EXPORT(iree_vmvx_mmt4d_f32f32f32, mmt4d, v) {
   IREE_TRACE_ZONE_BEGIN(z0);
-  // Here are abusing the 2D-specific macros MAP_BUFFER_2D_* to query 4D arrays
-  // in the sense of "query the outer 2 dimensions of this 4D array".
-  // All we need is the base pointer and the outer-most _stride0. This comes
-  // down to the fact that mmt4d requires the inner 3 dimensions of each 4D
-  // buffer operand to be contiguous row-major, as the whole point of mmt4d
-  // is to be able to assume an optimal memory layout.
-  // So far, mmt4d is always bufferized with all 4 dimensions of all
-  // 4D buffers contiguous row-major, but it's not a bad idea to keep the
-  // stride generality on the outer-most dimension (_stride0) as that could be
-  // handy for a few things (splitting the K dimension, extra alignment).
+  iree_host_size_t M = (iree_host_size_t)args->m;
+  iree_host_size_t N = (iree_host_size_t)args->n;
+  iree_host_size_t K = (iree_host_size_t)args->k;
+  iree_host_size_t M0 = (iree_host_size_t)args->m0;
+  iree_host_size_t N0 = (iree_host_size_t)args->n0;
+  iree_host_size_t K0 = (iree_host_size_t)args->k0;
+  iree_host_size_t lhs_tile_size = M0 * K0;
+  iree_host_size_t rhs_tile_size = N0 * K0;
+  iree_host_size_t out_tile_size = M0 * N0;
+
+  // Here are abusing the 2D-specific macros MAP_BUFFER_2D_* to query 4D arrays.
+  // Thanks to the requirement that all dimensions but the outer-most one are
+  // contiguous row-major, the outer-most stride is the only nontrivial stride,
+  // we can correctly coalesce the inner 3 dimensions without changing the
+  // mapped span.
   MAP_BUFFER_2D_RO(lhs, float,
                    /*buffer_ref=*/args->lhs_ref,
                    /*offset=*/args->lhs_offset,
                    /*stride0=*/args->lhs_row_stride,
                    /*stride1=*/1,
-                   /*size0=*/args->m,
-                   /*size1=*/args->k);
+                   /*size0=*/M,
+                   /*size1=*/K* lhs_tile_size);
   MAP_BUFFER_2D_RO(rhs, float,
                    /*buffer_ref=*/args->rhs_ref,
                    /*offset=*/args->rhs_offset,
                    /*stride0=*/args->rhs_row_stride,
                    /*stride1=*/1,
-                   /*size0=*/args->n,
-                   /*size1=*/args->k);
```

Comment:
surprised clang-format is ok with the missing space here (vs below where there's a space)!

Reply:
agree! I tried a few things, but we are clearly running into a clang-format bug here: it formats these two functions differently.